t/t2271-job-dependency-after.t

```diff
@@ -186,6 +186,12 @@ test_expect_success 'dependency=afternotok works' '
 	flux job wait-event -vt 15 \
 		-m type=dependency $ok1 exception
 '
+test_expect_success 'dependency=afternotok only applies to jobs that start' '
+	canceled_job=$(flux submit --urgency=hold false) &&
+	id2=$(flux submit --dependency=afternotok:$canceled_job hostname) &&
+	flux cancel $canceled_job &&
+	flux job wait-event -vt 15 -m type=dependency $id2 exception 
+'
```

Comment:
I guess it follows that a chain of dependent jobs would all be canceled but just for completeness, does it make sense to have an explicit test for it?

Reply:
Yes, probably. Good idea.

Reply:
Ok, I added a test that ensures a chain of afternotok are canceled when one is canceled, and another test that ensures the chain is canceled if a job succeeds.

Reply:
I also noticed some trailing whitespace, so added a commit to fix that.